app/imageswap/imageswap.py

```diff
@@ -89,6 +88,9 @@ def mutate():
 
     app.logger.debug(json.dumps(request.json))
 
+    # flag, whether there were any changes for at lease one image.
+    needs_patch = False
+
```

Comment:
question (blocking): Does this initialization/assignment actually need to be moved?

Reply:
Not really. I just wanted to keep the logic compact.
The initialization is directly before the block where at different places "needs_pacth" can be set to True.
"needs_patch" is a Flag whether there were any changes. So, in the logic block it can only be set to True. 

needs_patch = False
if ...
else
  for...
    if ...
      needs_patch = True
  if ...
  else ...
    needs_patch = True

But at the end, it doesn´t matter for me   :-)

Reply:
Moved variable creation back to the original position.